src/main/java/graphql/analysis/NodeVisitorWithTypeTracking.java

```diff
@@ -153,12 +154,12 @@ public TraversalControl visitField(Field field, TraverserContext<Node> context)
         boolean isTypeNameIntrospectionField = fieldDefinition == schema.getIntrospectionTypenameFieldDefinition();
         GraphQLFieldsContainer fieldsContainer = !isTypeNameIntrospectionField ? (GraphQLFieldsContainer) unwrapAll(parentEnv.getOutputType()) : null;
         GraphQLCodeRegistry codeRegistry = schema.getCodeRegistry();
-        Map<String, Object> argumentValues = ValuesResolver.getArgumentValues(codeRegistry,
+        Map<String, Object> argumentValues = !variables.isEmpty() ? ValuesResolver.getArgumentValues(codeRegistry,
                 fieldDefinition.getArguments(),
                 field.getArguments(),
                 CoercedVariables.of(variables),
                 GraphQLContext.getDefault(),
-                Locale.getDefault());
+                Locale.getDefault()) : new HashMap<>();
```

Comment:
I think the pattern to use here is Collections.emptyMap();

Reply:
Yup `emptyMap()` please